### PR TITLE
Add SECURITY.md

### PR DESCRIPTION
**Pull request by maurolacy**
Copied / adapted from [Osmosis security policy](https://github.com/osmosis-labs/osmosis/blob/main/SECURITY.md).

**TODO**:

- [x] Confirm general text / guidelines here.
- [x] Confirm security email.
- [x] Make [GitHub's Private Vulnerability Reporting](https://github.com/babylonchain/babylon/security/advisories/new) work ~~(or remove from policy)~~.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+| Severity     | Description                                                              |
+|--------------|--------------------------------------------------------------------------|
+| **CRITICAL** | Immediate threat to critical systems (e.g., chain halts, funds at risk). |
+| **HIGH**     | Significant impact on major functionality.                               |
+| **MEDIUM**   | Impacts minor features or exposes non-sensitive data.                    |
+| **LOW**      | Minimal impact.                                                          |
+
+## Bug Bounty
+
+Though we don't have an official bug bounty program, we generally offer rewards to security researchers who responsibly
```
Comment (vitsalis) [inline SECURITY.md:70]: @fishermanymc need to verify this sentence.

**Review by vitsalis — COMMENTED** [review]
Looks good to me in general.

@fishermanymc can you please also take a look to verify?

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+## Introduction
+
+Security researchers are essential in identifying vulnerabilities that may impact the Babylon ecosystem.
+If you have discovered a security vulnerability in the Babylon chain or any repository managed by Babylon,
+we encourage you to notify us using one of the methods outlined below.
+
+### Guidelines for Responsible Vulnerability Testing and Reporting
+
+1. **Refrain from testing vulnerabilities on our publicly accessible environments**, including but not limited to:
+  - Babylon mainnet.
```
[the remark this answers is not in the document]
Comment (fishermanymc) [reply SECURITY.md:12]: > Given that we don't have a mainnet yet, recommend we remove this and update later?

I think we should discourage testing using any of our public testnets.

File: SECURITY.md (hunk 1 above)
Comment (fishermanymc) [reply SECURITY.md:70]: we can say we are setting up one atm and will retro reward.

File: SECURITY.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (maurolacy) [reply SECURITY.md:12]: OK, will keep it as it is for the future.

**Comment by maurolacy** [comment]
@vitsalis or some admin, can you set the vulnerability reporting page? 

https://github.com/babylonchain/babylon/security/advisories/new

That, or we remove from policy.

**Review by vitsalis — APPROVED** [review]
Thanks for resolving the comments!

**Pull request merged**